File: leetcode/python/120.三角形最小路径和.py
class Solution:
    def minimumTotal(self, triangle: List[List[int]]) -> int:
        ret = 0
        n = len(triangle)
        if not triangle:
            return ret
        for i in range(n - 2, -1, -1):
            for j in range(i + 1):
                triangle[i][j] = min(triangle[i + 1][j], triangle[i + 1][j + 1]) + triangle[i][j]
        
        
        return triangle[0][0]

File: leetcode/golang/120.三角形最小路径和.go
func minimumTotal(triangle [][]int) int {
    if triangle == nil {
        return 0;
    }
    n := len(triangle);
    i := n - 2;
    for i >= 0 {
        j := 0;
        for j < i + 1 {
            a := triangle[i + 1][j];
            b := triangle[i + 1][j + 1];
            if a > b {
                triangle[i][j] += b;
            } else {
                triangle[i][j] += a;
            }
            j += 1;
        }
        i -= 1;
    }
    return triangle[0][0];
}